src/simulation/LoadGenerator.cpp

```diff
@@ -751,27 +718,76 @@ LoadGenerator::generateLoad(GeneratedLoadConfig cfg)
                     if (sorobanCfg.nWasms != 0)
                     {
                         --sorobanCfg.nWasms;
-                        return createUploadWasmTransaction(
-                            ledgerNum, sourceAccountId, cfg);
+
+                        auto wasm = cfg.modeSetsUpInvoke()
```

Comment:
nit: shall we factor out this and the following branches into separate methods to keep this function concise?